### PR TITLE
Merge 2.11.x to 2.12.x

### PR DESCRIPTION
**Pull request by retronym**
```
% export MB=$(git merge-base origin/2.12.x origin/2.11.x)
% echo $MB
2cfac1a3516850a40728fefd8f808613d2e83f84
% transcript git log --graph --oneline --decorate $MB...origin/2.11.x
https://gist.github.com/7c59590812a10f819070
% git merge eb15950
% git merge -sours 767cc44
% git merge 7ba38a0
% git merge -sours 421c599
% git merge origin/2.11.x
```

Review by @adriaanm. Please check changes to versions.properties and build.number.

These are the two `nomerge` commits that I skirted around with `-sours`.

```
| * | | | | | | 2c4ef6b (origin/pr/4196) [nomerge] SI-9030 don't call private BoxesRunTime.equalsNumChar

| * | | | | 38587c5 (origin/pr/4048) [nomerge] SI-8899 Revert "SI-8627 make Stream.filterNot non-eager"
```

### CODE REVIEW

**Comment by retronym** [comment]
PR Validation might not work until https://github.com/scala/scala-jenkins-infra/issues/16 is completed.

**Comment by retronym** [comment]
@adriaanm Please also review the last commit, 4176653.

File: build.number [hunk 1]
```diff
@@ -5,5 +5,5 @@ version.patch=0
 # This is the -N part of a version.  if it's 0, it's dropped from maven versions.
 version.bnum=0
 
-# Note: To build a release run ant with -Dbuild.release=true
-# To build an RC, run ant with -Dmaven.version.suffix=-RCN
+# To build a release, see scripts/jobs/scala-release-2.11.x-build
```
Comment (adriaanm) [inline build.number:8]: Purely FYI, this line is also obsolete in 2.11.x -- I removed version numbers from script names for this reason, but forgot to update the comment.

**Comment by adriaanm** [comment]
Quite the stress test for scabot on its first official day! I'll investigate why so many statuses are out of date tomorrow.

**Comment by retronym** [comment]
Most of the commits in the PR will predate the dawn of Scalabot. Plenty of the `xxx`-es come from

```
/tmp/hudson1067010863983445703.sh: line 2: scripts/jobs/validate/publish-core: No such file or directory
Build step 'Execute shell' marked build as failure
```

Was scabot even supposed to validate commits that were previously validated by the old validator?

image

**Comment by adriaanm** [comment]
Nope, the divide between old and new, botwise, is purely based on PR number. My disappointment was that I had to /synch to get the statuses to go from orange-? to red-X

**Comment by som-snytt** [comment]
Someone has been busy.

**Comment by adriaanm** [comment]
LGTM

**Comment by adriaanm** [comment]
I hope there won't be too much rebasing in the other 2.12.0-M1 PRs' futures.

**Pull request merged**